projects/kit/components/files/input-files/input-files.directive.ts

```diff
@@ -54,11 +54,13 @@ export class TuiInputFilesDirective
     public readonly input = tuiInjectElement<HTMLInputElement>();
 
     public process(files: FileList): void {
-        this.onChange(
-            this.input.multiple
-                ? [...toArray(this.value()), ...Array.from(files)]
-                : files[0] || null,
-        );
+        const fileOrFiles = this.input.multiple
+            ? [...toArray(this.value()), ...Array.from(files)]
+            : files[0] || null;
```

Comment:
No need to fallback to `null` if we check below

Reply:
fixed